src/mrpro/algorithms/reconstruction/_direct_reconstruction.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import annotations
+
+from typing import Self
+
+from mrpro.algorithms._prewhiten_kspace import prewhiten_kspace
+from mrpro.algorithms.reconstruction import Reconstruction
+from mrpro.data._CsmData import CsmData
+from mrpro.data._DcfData import DcfData
+from mrpro.data._IData import IData
+from mrpro.data._kdata._KData import KData
+from mrpro.data._KNoise import KNoise
+from mrpro.operators._FourierOp import FourierOp
+from mrpro.operators._LinearOperator import LinearOperator
+
+
+class DirectReconstruction(Reconstruction):
+    """Direct Reconstruction by Inverse Fourier Transform."""
+
+    def __init__(
+        self,
+        fourier_operator: LinearOperator,
+        csm: None | CsmData = None,
```

Comment:
I would use csmdata, dcfdata and knoise to higlight that these are dataclasses

Reply:
this is already defined by the type?

Reply:
yes but later in the code or if you call DirectReconstruction.csm you don't know if csm is now csmdata or csm_tensor.